moveos/moveos-types/src/object.rs

```diff
@@ -129,98 +132,120 @@ pub type SequenceNumber = u64;
 
 #[derive(Eq, PartialEq, Debug, Clone, Deserialize, Serialize, Hash)]
 pub struct Object {
+    //TODO should contains self id.
+    //pub id: ObjectID,
     pub data: ObjectData,
     /// The owner that unlocks this object
     pub owner: Owner,
 }
 
 impl Object {
-    pub fn new_table_object(object: TableObject) -> Self {
+    pub fn new_table_object(value: TableObjectData) -> Self {
         Self {
-            data: ObjectData::TableObject(object),
+            data: ObjectData::Table(value),
             //TODO: set the owner
             owner: Owner::Immutable,
         }
     }
 
-    pub fn new_move_object(object: MoveObject) -> Self {
+    pub fn new_move_object(value: MoveObjectData) -> Self {
         Self {
-            data: ObjectData::MoveObject(object),
+            data: ObjectData::Move(value),
             //TODO: set the owner
             owner: Owner::Immutable,
         }
     }
+
+    pub fn as_move_resource(&self, id: ObjectID) -> Result<Vec<u8>> {
```

Comment:
I think the move_resource will be a struct(type), so the funtions name`as_move_resource`  will like  `as_resource_bytes`

Reply:
This is a draft method. I try to find a way to map the Move Object struct to Rust Object. Because the Move Object has a generic type parameter `T`, it is hard to implement in Rust.

Reply:
Why need this map( Move object to Rust object)?

Reply:
We need parse the Move object in Rust code.

Reply:
I rename this function to `as_object_argument`

Reply:
> I think there's no need to parse the whole Object. Maybe parse the Object::id field is enough?

Yes.

Rust Move Object
```
struct MoveObject{
  id: ObjectID,
  value: Vec<u8>,
}
```

Move Object
```
struct Object<T>{
   id: address,
   value: T,
}
```

I can not find a simple method to make rust `bcs::to_bytes(&move_object)` == move `bcs::to_bytes(&object)`